pkgs/development/compilers/cudatoolkit/common.nix

```diff
@@ -191,7 +209,7 @@ stdenv.mkDerivation rec {
   preFixup =
     let rpath = lib.concatStringsSep ":" [
       (lib.makeLibraryPath (runtimeDependencies ++ [ "$lib" "$out" "$out/nvvm" ]))
-      "${stdenv.cc.cc.lib}/lib64"
+      "${gcc.cc.lib}/lib64"
```

Comment:
the gcc/stdenv distinction here is subtle enough i believe it deserves a comment

Reply:
How about this one?

Reply:
I guess my confusion was rather why reference `gcc` directly instead of accessing it through `stdenv`

Reply:
Oh, right. So, I should explain that `gcc` here is what's we override in `extension.nix` based on `versions.toml`
Actually, maybe I should override `stdenv` too? Like so

https://github.com/SomeoneSerge/nixpkgs/blob/cc4f01552c2dca50b452170df2770edb71148555/pkgs/development/compilers/cudatoolkit/extension.nix#L11-L15

Reply:
ooh yeah not a bad idea...

Reply:
could we get away with only overriding `stdenv` and then pulling the gcc version from that `stdenv`?

Reply:
@samuela Good. I'm thinking about exposing that `stdenv` in `cudaPackages` (rather than `cudatoolkit`) then. I, however, feel uneasy about exposing it as `cudaPackages.stdenv` because it might affect people's expectations... E.g. since `clangStdenv` contains `clang`, people might think `cudaPackages.stdenv` contains `nvcc`

Reply:
Alt names I'm thinking of: `cudaStdenv` (might be misinterpreted the same way), `backendStdenv` (exactly what it is, but hard to pronounce 🤣 ). Do you like any?

Reply:
Good point! I agree that `cudaStdenv` could be slightly misleading. It's a little tricky to name... Maybe `matchingStdenv`? `compatibleStdenv`? Idk I'm happy with whatever you feel is most appropriate

Reply:
I kept the ugly name, because "backend for nvcc" seemed like the clearest description...